components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt

```diff
@@ -102,6 +121,13 @@ class DefaultTopSitesStorage(
             try {
                 providerTopSites = topSitesProvider
                     .getTopSites(allowCache = true)
+                    .filter {
```

Comment:
I don't think this needs to be in AC. We only ever get at most 2 contile sites so even if it's ebay or Amazon. We wouldn't get anymore additional sites. This is also a generic implementation of top sites storage so it shouldn't need to know any specific handling for Fenix.

Reply:
Thank you for your comment, the initial idea I had was to add a generic `(TopSite)->Boolean` function inside `TopSitesProviderConfig` and use it to filter the provided sponsored top sites, as opposed to filtering the received list of top sites in Fenix. My implementation obviously went the wrong way from here, but since the client provides this configuration, could we consider using it?